### PR TITLE
Use decimal numbers for endpoint/execution IDs

### PR DESCRIPTION
**Pull request by michael-o**
As discussed some time ago on the dev mailing list.

### CODE REVIEW

File: httpcore5/src/main/java/org/apache/hc/core5/reactor/IOSessionImpl.java [hunk 1]
```diff
@@ -71,7 +71,7 @@ public IOSessionImpl(final String type, final SelectionKey key, final SocketChan
         this.commandQueue = new ConcurrentLinkedDeque<>();
         this.lock = new ReentrantLock();
         this.socketTimeout = Timeout.DISABLED;
-        this.id = String.format(type + "-%08X", COUNT.getAndIncrement());
+        this.id = String.format(type + "-%08d", COUNT.getAndIncrement());
```
Comment (ok2c) [inline httpcore5/src/main/java/org/apache/hc/core5/reactor/IOSessionImpl.java:74]: @michael-o `08d` would not be enough for a LONG value, would it? Should not it be 20?

Comment (carterkozak) [reply httpcore5/src/main/java/org/apache/hc/core5/reactor/IOSessionImpl.java:74]: If the padding isn't important and we're okay with ambiguous substring matches, it could be worth considering simple concatenation to avoid poor String.format performance until we get something like [jep-348](https://openjdk.java.net/jeps/348).
```suggestion
        this.id = type + "-" + COUNT.getAndIncrement();
```

Comment (michael-o) [reply httpcore5/src/main/java/org/apache/hc/core5/reactor/IOSessionImpl.java:74]: @ok2c I agree in general. I think we need to add 60%, why 16 (base)/10 (base) = 1,6. So 13 or 14. I am also OK with 20 which, though, seems quite long to me.

@carterkozak I can live with both. Whatever the community seems better readable/greppable.

Comment (ok2c) [reply httpcore5/src/main/java/org/apache/hc/core5/reactor/IOSessionImpl.java:74]: > @ok2c I agree in general. I think we need to add 60%, why 16 (base)/10 (base) = 1,6. So 13 or 14. I am also OK with 20 which, though, seems quite long to me.

@michael-o Is not `9,223,372,036,854,775,807`longest possible LONG?

Comment (michael-o) [reply httpcore5/src/main/java/org/apache/hc/core5/reactor/IOSessionImpl.java:74]: > 
> 
> > @ok2c I agree in general. I think we need to add 60%, why 16 (base)/10 (base) = 1,6. So 13 or 14. I am also OK with 20 which, though, seems quite long to me.
> 
> @michael-o Is not `9,223,372,036,854,775,807`longest possible LONG?

That's correct. I wanted to make a compromise. Having `ex-00000000000000000001` looks awkward.

Comment (ok2c) [reply httpcore5/src/main/java/org/apache/hc/core5/reactor/IOSessionImpl.java:74]: @michael-o Fair enough.

Comment (carterkozak) [reply httpcore5/src/main/java/org/apache/hc/core5/reactor/IOSessionImpl.java:74]: Another option is to exclude padding in favor of a trailing separator for concise identifiers that cannot be ambiguous:

```java
this.id = type + "[" + COUNT.getAndIncrement() + "]";
```

This solves the problem grepping between `ex-1` and `ex-10` by producing values `ex[1]` and `ex[10]`, and in my opinion is easier for humans to quickly parse without N leading zero characters. What do you think?

Comment (michael-o) [reply httpcore5/src/main/java/org/apache/hc/core5/reactor/IOSessionImpl.java:74]: @carterkozak While you are right, it should read as an array. That's the whole point.

**Comment by michael-o** [comment]
My counter proposal is to follow @ok2c and increase to 10. This makes by default possible to fit multiples of `Integer.MAX_VALUE` and still retain a not too long string. Everything else will need to scale. Having no padding as @carterkozak proposed requires less operations in the JVM, but the will leave logging like a flush left text, hard to follow with the eye.
